docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,132 @@
+## Backpack-android architecture overview:
+
+Backpack-android is a Android a traditional gradle Android lib project, composed of two modules.
+
+- app: The example app
+- Backpack: The Backpack Android components
+
+## App
+
+### Stories
+
+Stories are defined programmatically in tree like structure that is then used to build the UI.
```

Comment:
```suggestion
Stories are defined programmatically in dendroidal structure that is used to build the UI.
```

Reply:
I have no idea what `dendroidal` means lol. I'll use @shaundon suggestion instead as I think is easier to understand.